File: Java/Medium/RuntimeError/maximum-number-of-groups-entering-a-competition_4.java
class Solution {
    public int maximumGroups(int[] grades) {
        
        int[] counts = new int[100001];
        for (int grade : grades) {
            counts[grade]++;
        }
        int ans = 0;
        for (int i = 1; i < counts.length; i++) {
            ans += counts[i] / i;
            counts[i + 1] += counts[i] % i;
        }
        return ans;
    }
}

File: Rust/Medium/WrongAnswer/maximum-number-of-groups-entering-a-competition_4.rs
impl Solution {
    pub fn maximum_groups(grades: Vec<i32>) -> i32 {
        
        let mut groups = vec![];
        for grade in grades {
            let mut i = 0;
            while i < groups.len() {
                if groups[i] >= grade {
                    break;
                }
                i += 1;
            }
            if i < groups.len() {
                groups[i] = grade;
            } else {
                groups.push(grade);
            }
        }
        groups.len() as i32
    }
}